code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushTemplate.java

```diff
@@ -0,0 +1,418 @@
+        }
+
+        try {
+            this.deliveryId = DataReader.getString(data, CampaignPushConstants.Tracking.Keys.DELIVERY_ID);
+        } catch (final DataReaderException dataReaderException) {
+            throw new IllegalArgumentException("Required field \"_dId\" not found.");
+        }
+
+        // optional push template data
+        this.payloadVersion = Integer.parseInt(DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.VERSION, CampaignPushConstants.DefaultValues.LEGACY_PAYLOAD_VERSION_STRING));
+        this.sound = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.SOUND, "");
+        this.imageUrl = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.IMAGE_URL, "");
+        this.channelId = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.CHANNEL_ID, "");
+        this.actionUri = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.ACTION_URI, "");
+        this.icon = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.ICON, "");
+        this.expandedBodyText = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.EXPANDED_BODY_TEXT, "");
+        this.expandedBodyTextColor = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.EXPANDED_BODY_TEXT_COLOR, "");
+        this.titleTextColor = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.TITLE_TEXT_COLOR, "");
+        this.smallIconColor = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.SMALL_ICON_COLOR, "");
+        this.notificationBackgroundColor = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.NOTIFICATION_BACKGROUND_COLOR, "");
+
+        try {
+            final String count = data.get(CampaignPushConstants.PushPayloadKeys.BADGE_NUMBER);
+            if (count != null) {
+                this.badgeCount = Integer.parseInt(count);
+            }
+        } catch (final NumberFormatException e) {
+            Log.debug(CampaignPushConstants.LOG_TAG, SELF_TAG, "Exception in converting notification badge count to int - %s", e.getLocalizedMessage());
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            this.notificationImportance = getNotificationImportanceFromString(data.get(CampaignPushConstants.PushPayloadKeys.NOTIFICATION_PRIORITY));
+        } else {
+            this.notificationPriority = NotificationPriority.from(data.get(CampaignPushConstants.PushPayloadKeys.NOTIFICATION_PRIORITY));
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            this.notificationVisibility = getNotificationVisibilityFromString(data.get(CampaignPushConstants.PushPayloadKeys.NOTIFICATION_VISIBILITY));
+        }
+
+        this.actionType = getActionTypeFromString(data.get(CampaignPushConstants.PushPayloadKeys.ACTION_TYPE));
+        this.actionButtons = getActionButtonsFromString(data.get(CampaignPushConstants.PushPayloadKeys.ACTION_BUTTONS));
+    }
+
```

Comment:
Where applicable, can you please annotate these getters with `@NonNull` or `@Nullable` so that the callers can be aware during compile time via warning to do a null check.